clippy_lints/src/semicolon_block.rs

```diff
@@ -135,3 +183,45 @@ fn semicolon_outside_block(cx: &LateContext<'_>, block: &Block<'_>, tail_stmt_ex
         },
     );
 }
+
+fn semicolon_outside_block_if_singleline(
+    cx: &LateContext<'_>,
+    block: &Block<'_>,
+    remove_span: Span,
+    insert_span: Span,
+    inequality: bool,
+    ty: &str,
+) {
+    let (remove_line, insert_line) = (get_line(cx, remove_span), get_line(cx, insert_span));
+
+    let eq = if inequality {
+        remove_line != insert_line
+    } else {
+        remove_line == insert_line
+    };
+
+    if eq {
+        span_lint_and_then(
+            cx,
+            SEMICOLON_OUTSIDE_BLOCK_IF_SINGLELINE,
+            block.span,
+            &format!("consider moving the `;` {ty} the block for consistent formatting"),
+            |diag| {
+                multispan_sugg_with_applicability(
+                    diag,
+                    "put the `;` here",
+                    Applicability::MachineApplicable,
+                    [(remove_span, String::new()), (insert_span, ";".to_owned())],
+                );
+            },
+        );
+    }
+}
+
+fn get_line(cx: &LateContext<'_>, span: Span) -> usize {
+    cx.sess()
+        .source_map()
+        .lookup_line(span.lo())
+        .expect("failed to get span's line")
+        .line
```

Comment:
This `expect` looks a bit suspicious, I believe that there are some error cases for macros and there might be others. Generally speaking, it's advisable to handle errors to avoid panics. I think the simplest way would be to simply to not lint if this returns an error.

It can be okay to use `expect()` in clippy, but in that case, it should be explained, why this is safe. The expect doesn't seem to do that in this case.

Reply:
Wasn't sure what to do here, it doesn't lint over panicking now.